x-pack/plugins/stack_connectors/server/connector_types/lib/servicenow/service.ts

```diff
@@ -249,6 +255,72 @@ export const createExternalService: ServiceFactory = ({
     }
   };
 
+  const getIncidentByCorrelationId = async (correlationId: string): Promise<ServiceNowIncident> => {
+    try {
+      const res = await request({
+        axios: axiosInstance,
+        url: getIncidentByCorrelationIdUrl(correlationId),
+        method: 'get',
+        logger,
+        configurationUtilities,
+      });
+
+      checkInstance(res);
+
+      if (res.status === 404) {
+        logger.warn(`No incident found with correlation ID ${correlationId}.`);
+      }
+
+      const foundIncident = res.data.result[0] ?? null;
+
+      return foundIncident;
+    } catch (error) {
+      throw createServiceError(error, `Unable to get incident by correlation ID ${correlationId}`);
+    }
+  };
+
+  const closeIncident = async (params: ExternalServiceParamsClose) => {
+    try {
+      const { correlationId, incidentId } = params;
+      let incidentToBeClosed = null;
+
+      if (correlationId == null && incidentId == null) {
+        throw new Error('No correlationId or incidentId found.');
+      } else if (incidentId) {
+        incidentToBeClosed = await getIncident(incidentId);
```

Comment:
This could throw a 404 error, right?

Reply:
Updated in [2cdac29](https://github.com/elastic/kibana/pull/171760/commits/2cdac29775d8ca3b3989ceb667e623ec9a6b033e)

Reply:
updated in https://github.com/elastic/kibana/pull/171760/commits/7088a167bdb1d3ab97f992bb0274cca8d57d489d